### ISSUE TITLE
Sometimes event callbacks are unexpectedly triggered twice

### ISSUE THREAD

**Issue by alicepolice** [zMoooooritz/go-let-loose#2]
The two event callbacks were triggered with a 0.8-minute interval.
![image](https://github.com/user-attachments/assets/d54fc806-f667-43c8-be01-de5527b42cee)
![image](https://github.com/user-attachments/assets/775506b5-87ef-4267-a637-f01996064319)

**Comment by zMoooooritz** [comment]
Thank you for the report, unfortunately I am not sure what I should see in these two images? Since I have no clue what that UI does

For the first one I assume you receive the MatchEndEvent two times? If this is the case I would have to take a closer look
Concerning the second one you need to elaborate a bit more please

**Comment by alicepolice** [comment]
These two images together indicate that the MATCH END event callback was executed twice. The first image shows that the match information was saved twice, while the second image shows that the player information was saved twice. I discovered this by chance; the issue occurred after saving around dozens of matches.

```
type MatchEnd struct{}

func (p *MatchEnd) IsSubscribed(e event.Event) bool {
	if e.Type() == event.EVENT_MATCHEND {
		return true
	} else {
		return false
	}

func (p *MatchEnd) Notify(e event.Event) {
	v, ok := e.(event.MatchEndEvent)
	if ok {
		// Retrieve the current match and player data and save it to the database.
	}
}
```

Additionally, there's another issue that's quite frustrating. I designed a command for quick suicide, but occasionally, the server executes the 'punish' command and still reports a timeout, causing the command to be sent and executed again. This results in the player being forced to suicide twice after respawning, essentially turning one 'kkkk' command into two suicides. From my observation, the server has already executed the 'punish' command for those who issued the 'kkkk' command, so there's no need to re-execute it. However, the program doesn't seem to recognize this.

```
if strings.Contains(v.Message, "kkkk") || strings.Contains(v.Message, "KKKK") {
	fmt.Println("[聊天指令] ", v.Player, "打了kkkk")
	rcn.PunishPlayer(v.Player.Name, "KKKK！已经帮您重生辣！正在重生中~")
} 
```

<img width="374" alt="3488cb428eb11a4ab271f86d41121a45" src="https://github.com/user-attachments/assets/760af1fd-94f2-42b1-835c-30b688ed5c48">
<img width="371" alt="9727b220225c78699db46bba364a1fec" src="https://github.com/user-attachments/assets/29084664-eed1-4e26-98e7-4c6dcded3ef1">


My current solution is to create a connection that uses a proxy. The proxy helps reduce the frequency of timeouts, but it cannot completely eliminate the issue, as timeouts still occur.

Finally, the tool you created is excellent. Thank you.

**Comment by zMoooooritz** [comment]
Okay, thanks for clarifying I have so far no clue how these problems arise

1. Could you please trace the MatchEndEvent as is, that you receive in Notify (in order to verify, that it is indeed originates from the same "logline")

2. Please provide me with more logs (before and after those timeouts occour). It would be best to have them as raw text (you could share them with [pastebin](https://pastebin.com/) or one of its alternatives)

With those traces I should be able to narrow down the issue

Thank you for reporting those issues and your positive feedback!

**Comment by zMoooooritz** [comment]
I have just resolved the first issue in the commit 1e5103b (which is included in v0.0.6 of the module)

The error did indeed cause events to be issued multiple times in specific circumstances (independend of the event type)

Potentially this also solves your second issue, I will close this issue for now

Feel free to open a new issue as soon as this or another problem pops up
